File: leetcode_py/p2496.py
from typing import List


class Solution:
    def maximumValue(self, strs: List[str]) -> int:
        # def get_len(s: str) -> int:
        #     # 数字
        #     for s in str:
        #         if not s.isdigit():
        #             return len(s)
        #     return int(s)
        # max_len = 0

        # for str in strs:
        #     max_len = max(get_len(str),max_len)
        # return max_len
        res  =0
        for s in strs:
            is_digits = all(c.isdigit() for c in s)
            res = max(res,int(s) if is_digits else len(s))
        return res
            

File: leetcode_C++/p2496.cpp
/*
@Author: mrxhar
@Date:   2023-09-06 08:45:12
@Email:  1337904145@qq.com
@Blog:   https://leetcode.cn/problems/maximum-value-of-a-string-in-an-array/description/?envType=daily-question&envId=2023-09-06
*/
#include <vector>
using std::vector;
#include <string>
using std::string;
#include <cmath>
using std::max;

class Solution {
public:
    int maximumValue(vector<string>& strs) {
        int res = 0;
        for (auto str : strs) {
            bool isDigits = true;
            for (auto c : str) {
                if (!isdigit(c)) {
                    isDigits = false;
                    break;
                }
            }
            res = max(res, (int)(isDigits ? stoi(str) : str.length()));
        }
        return res;
    }
};

int main() {
    return 0;
}
